pip_run/commands.py

```diff
@@ -74,6 +75,38 @@ def separate(args):
     return _separate_script(args)
 
 
+def infer_ipython(sep_args):
+    """
+    Check for the presence of the argument 'ipython' in pip_args.
+
+    If present and py_args is empty, return arguments for pip and
```

Comment:
I'm unsure whether "py_args is empty" is a worthwhile exclusion. What if you wanted to run a script in interactive mode with IPython:

```
pip-run ipython -- -i script.py
```

We can consider expanding to support that use-case in the future.